04_design_patterns/src/shipments/ShipmentDecorator.ts

```diff
@@ -0,0 +1,40 @@
+export class ShipmentMarksDecorator {
+    constructor(private child: any, private marks: string[]) {
+    }
+
+    public ship() {
+        const marks = this.getFormattedMarksToApply();
+
+        return `${this.child.ship()}${ marks.length > 0 ? `\n${this.getFormattedMarksToApply()}` : ''}`
+    }
+
+    private getFormattedMarksToApply() {
+        return this.marks.reduce(
+            (acc: string[], mark: string) => {
+                if (mark === 'Fragile') {
+                    return [
+                        ...acc,
+                        '**MARK FRAGILE**'
+                    ]
+                }
+
+                if (mark === 'Do Not Leave') {
+                    return [
+                        ...acc,
+                        '**MARK DO NOT LEAVE IF ADDRESS NOT AT HOME**'
+                    ]
+                }
+
+                if (mark === 'Return Receipt Requested') {
+                    return [
+                        ...acc,
+                        '**MARK RETURN RECEIPT REQUESTED**'
+                    ]
+                }
+
+                return acc;
+            },
+            []
+        );
+    }
```

Comment:
Nice!
Since we have some time left, you could try to implement this part with a typescript decorator.